Assets/Scripts/SimpleCoin.cs

```diff
@@ -5,13 +5,46 @@ namespace DefaultNamespace
 {
     public class SimpleCoin : MonoBehaviour
     {
+        [SerializeField] private int _value = 1;
         [SerializeField] private string _playerTag = "Player";
+        [SerializeField] private float _movementTime = 1f;
+        [SerializeField] private float _movementSpeed = 1f;
+        [SerializeField] private float _rotationsSpeed = 1f;
         private void OnTriggerEnter(Collider other)
         {
             if (other.CompareTag(_playerTag))
             {
+                _coinCounter.Add(_value);
                 Destroy(gameObject);
             }
         }
+
+        private void Start()
+        {
+            _rigidbody = GetComponent<Rigidbody>();
+            _coinCounter = GameObject.FindWithTag("GameController").GetComponent<CoinCounter>();
```

Comment:
Relying on tags for object search is too error-prone. Also, there are many coins and this global search can take some time. 
Solution:
- Inject CoinCounter when instantiating a coin (e.g., via method)